### PR TITLE
Add caddy2 configuration file

### PR DESCRIPTION
**Pull request by rjpruitt16**
https://github.com/syncthing/syncthing/issues/6992

### CODE REVIEW

**Review by hildebro — APPROVED** [review]
Can confirm this works on a clean installation.

**Comment by calmh** [comment]
I think for this to be useful it should at minimum map the UI to a subpath like /syncthing, as the Caddy 1 example does. It should maybe also not set the port to 2016.

**Comment by rjpruitt16** [comment]
Should the port be deleted. I thought it would convey more information if set. I could also put example.com.

**Comment by rjpruitt16** [comment]
:)

**Comment by imsodin** [comment]
> Should the port be deleted. I thought it would convey more information if set. I could also put example.com.

I don't know caddy, but looking at the example for v1 and Jakobs response above: Yes, to get it in line with the other example.

Also there's now two identical "Caddy" subtitles. Either put both examples under one subtitle and specify them as v1 and v2 in text or have "Caddy v1" and "Caddy v2" (except that Caddy might not use "v1" notation but something else, use whatever is common).

**Comment by rjpruitt16** [comment]
any more feedback

File: users/reverseproxy.rst [hunk 1]
```diff
@@ -68,6 +68,13 @@ Caddy
         write none
         header none
     }
+    
+    caddy2
```
Comment (imsodin) [inline users/reverseproxy.rst:72]: That's part of the code block - probably shouldn't be, right?

Comment (rjpruitt16) [reply users/reverseproxy.rst]: I wanted to follow your advice of combining caddy into 1 subtitle. I took a second look, and it did look a little weird. I separated them into different subtitles

**Pull request merged**